src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -2200,6 +2204,8 @@ public WorkspacePreferences getWorkspacePreferences() {
         EasyBind.listen(workspacePreferences.showAdvancedHintsProperty(), (obs, oldValue, newValue) -> putBoolean(SHOW_ADVANCED_HINTS, newValue));
         EasyBind.listen(workspacePreferences.warnAboutDuplicatesInInspectionProperty(), (obs, oldValue, newValue) -> putBoolean(WARN_ABOUT_DUPLICATES_IN_INSPECTION, newValue));
         EasyBind.listen(workspacePreferences.confirmDeleteProperty(), (obs, oldValue, newValue) -> putBoolean(CONFIRM_DELETE, newValue));
+        workspacePreferences.getSelectedSlrCatalogs().addListener((ListChangeListener<String>) change ->
```

Comment:
Oh, wow. Did not know that the code really needs to be that different to the other code the lines above.

The code snipped was taken from org.jabref.preferences.JabRefPreferences#getGuiPreferences.

Reply:
I think the discussion is worth continuing in case we have to give reference of this PR to someone working on preferences in future.

By the snipped code I suppose you mean the lines (2204-2207) framed above, in #`getWorkspacePreferences` (not #`getGuiPrefrences`)? Or were you going to attach a separate snippet?

Yes, if you look at how `putStringList` is handled in the rest of the preferences, we don't use EasyBind there (as we would need to have a "property" getter for that):
https://github.com/JabRef/jabref/blob/833cf30f7eae5405e1da9a02e5db3b7b07247c0c/src/main/java/org/jabref/preferences/JabRefPreferences.java#L2968-L2974
This is #`getAutoCompletePreferences` - notice the odd one out here - in the case of StringList.

There is however, just one exception, in case of key bindings (#`getKeyBindingRepository`), where a property is indeed returned by design (`MapProperty<KeyBinding, String>`, to be precise), and hence, could use EasyBind:
https://github.com/JabRef/jabref/blob/833cf30f7eae5405e1da9a02e5db3b7b07247c0c/src/main/java/org/jabref/preferences/JabRefPreferences.java#L1257-L1260

In our case, I chose to follow the norm rather than the exception, but technically we could return a `MapProperty<String, Boolean>` for <CatalogName, EnabledStatus> and thus use EasyBind to achieve the same result with code that looks less out of place (we would then be handling two sets of values together - more complexity instead of the way I made it one-dimensional (just store the enabled catalogs)).

Reply:
Update: I find that there is `ListProperty` that could be used. This takes care of the complexity.
Is this PR okay or should I give it a shot using EasyBind?

Reply:
You don't need listproperty here, the normal change listener is fine